rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/reporting/rules/CreateDependencyGraphReportRuleProvider.java

```diff
@@ -0,0 +1,96 @@
+import org.jboss.windup.config.metadata.RuleMetadata;
+import org.jboss.windup.config.operation.GraphOperation;
+import org.jboss.windup.config.phase.ReportGenerationPhase;
+import org.jboss.windup.graph.GraphContext;
+import org.jboss.windup.graph.model.ProjectModel;
+import org.jboss.windup.graph.model.resource.FileModel;
+import org.jboss.windup.graph.service.WindupConfigurationService;
+import org.jboss.windup.reporting.model.ApplicationReportModel;
+import org.jboss.windup.reporting.model.TemplateType;
+import org.jboss.windup.reporting.service.ApplicationReportService;
+import org.jboss.windup.reporting.service.ReportService;
+import org.jboss.windup.rules.apps.java.condition.SourceMode;
+import org.ocpsoft.rewrite.config.Configuration;
+import org.ocpsoft.rewrite.config.ConfigurationBuilder;
+import org.ocpsoft.rewrite.context.EvaluationContext;
+
+import com.google.common.collect.Iterables;
+
+import java.util.List;
+
+/**
+ * Generates the dependency graph reports for the applications analyzed
+ */
+@RuleMetadata(phase = ReportGenerationPhase.class)
+public class CreateDependencyGraphReportRuleProvider extends AbstractRuleProvider
+{
+   public static final String REPORT_NAME = "Dependencies Graph";
+   public static final String REPORT_DESCRIPTION = "This graph shows the dependencies embedded within the analyzed application.";
+   public static final String TEMPLATE = "/reports/templates/dependency_graph.ftl";
+   private static final String REPORT_BASEFILENAME = "dependency_graph_report";
+
+   @Override
+   public Configuration getConfiguration(RuleLoaderContext ruleLoaderContext)
+   {
+      return ConfigurationBuilder.begin()
+               .addRule()
+               .when(SourceMode.isDisabled())
+               .perform(new CreateDependencyGraphReportOperation());
+   }
+
+   private class CreateDependencyGraphReportOperation extends GraphOperation
+   {
+
+      @Override
+      public void perform(GraphRewrite event, EvaluationContext context)
+      {
+         List<FileModel> inputPaths = WindupConfigurationService.getConfigurationModel(event.getGraphContext())
+                  .getInputPaths();
+         if (Iterables.size(inputPaths) > 1)
+         {
+            createGlobalAppDependencyGraphReport(event.getGraphContext());
+         }
+         inputPaths.stream()
+                  .filter(inputPath -> !ProjectModel.TYPE_VIRTUAL.equals(inputPath.getProjectModel().getProjectType()))
+                  .forEach(inputPath -> createSingleAppDependencyGraphReport(event.getGraphContext(),
+                           inputPath.getProjectModel()));
+      }
+
+      private ApplicationReportModel createAppDependencyGraphReport(GraphContext context)
+      {
+         ApplicationReportService applicationReportService = new ApplicationReportService(context);
+         ApplicationReportModel report = applicationReportService.create();
+         report.setReportPriority(104);
```

Comment:
Perhaps the '104' could be a constant?

Reply:
It could but in this case it's a "one time" usage just to set this value and there's no further need to reference this value somewhere else in the code so i think it's anyway fine to have it define in this way.